crates/router/src/core/payments/customers.rs

```
@@ -72,23 +74,29 @@ pub async fn create_connector_customer<F: Clone, T: Clone>(
     Ok(connector_customer_id)
 }
 
-pub fn get_connector_customer_details_if_present<'a>(
-    customer: &'a domain::Customer,
+pub fn get_connector_customer_details_if_present(
+    customer: &domain::Customer,
     connector_name: &str,
-) -> Option<&'a str> {
+) -> Option<String> {
     customer
         .connector_customer
         .as_ref()
-        .and_then(|connector_customer_value| connector_customer_value.get(connector_name))
-        .and_then(|connector_customer| connector_customer.as_str())
+        .and_then(|connector_customer_value| {
+            connector_customer_value
+                .clone()
+                .expose()
+                .get(connector_name)
+                .cloned()
+        })
+        .and_then(|connector_customer| serde_json::from_value::<String>(connector_customer).ok())
```

we should not ignore the error here